module/os/linux/zfs/zpl_file_range.c

```diff
@@ -179,8 +187,14 @@ zpl_clone_file_range(struct file *src_file, loff_t src_off,
 	if (len == 0)
 		len = i_size_read(file_inode(src_file)) - src_off;
 
-	return (__zpl_clone_file_range(src_file, src_off,
-	    dst_file, dst_off, len));
+	/* The entire length must be cloned or this is an error. */
+	ssize_t ret = zpl_clone_file_range_impl(src_file, src_off,
+	    dst_file, dst_off, len, zfs_bclone_wait_dirty);
+
+	if (len != ret)
```

Comment:
This should preserve other errors (`ret < 0` case). Add check for `ret >= 0`?

Same for `zpl_remap_file_range`

Reply:
Fixed.